src/target/ext/exec_file.rs

```diff
@@ -13,7 +14,13 @@ pub trait ExecFile: Target {
     /// process running on the remote system.
     /// If no `pid` is provided, return the filename corresponding to the
     /// currently executing process.
-    fn get_exec_file(&self, pid: Option<Pid>) -> TargetResult<&[u8], Self>;
+    fn get_exec_file<'a>(
+        &self,
+        pid: Option<Pid>,
+        offset: <Self::Arch as Arch>::Usize,
+        length: <Self::Arch as Arch>::Usize,
+        buf: &'a mut [u8],
+    ) -> TargetResult<&'a [u8], Self>;
```

Comment:
instead of returning the slice that was written into, we should follow standard Rust conventions and return a `usize` corresponding to the number of bytes written into `buf`.

you find this pattern in most similar APIs, most notably the `std::io::Read::read` API.